src/main/java/org/magicdgs/thaplv/haplotypes/model/AbstractHaplotypeConverter.java

```diff
@@ -0,0 +1,173 @@
+import java.util.Set;
+import java.util.stream.Collectors;
+
+/**
+ * Abstract implementation of {@link HaplotypeConverter}, with shared code for different {@link
+ * HaplotypeModel}
+ *
+ * @author Daniel Gomez-Sanchez (magicDGS)
+ */
+public abstract class AbstractHaplotypeConverter implements HaplotypeConverter {
+
+    /**
+     * The final missing allele for the final ploidy of the haplotype
+     */
+    protected final List<Allele> missingAllele;
+
+    /**
+     * Construct an abstract converter. The ouptut ploidy indicates the ploidy for the returned
+     * {@link VariantContext} for the converter.
+     */
+    protected AbstractHaplotypeConverter(final int outputPloidy) {
+        Utils.validateArg(outputPloidy >= 1, "outputPloidy < 1");
+        missingAllele = Collections.nCopies(outputPloidy, Allele.NO_CALL);
+    }
+
+    /**
+     * Get the output ploidy for the converter.
+     */
+    public int getOutputPlotidy() {
+        return missingAllele.size();
+    }
+
+    /**
+     * Is this genotype considered missing by this model? It should return always true if the
+     * genotype is completely missing.
+     *
+     * Behaviour of this method may be unexpected for non-diploid individuals in implementations.
+     *
+     * @param genotype the genotype to test
+     *
+     * @return {@code true} if it is considered missing; {@code false} otherwise
+     */
+    abstract boolean isConsideredMissing(final Genotype genotype);
+
+    /**
+     * Get the haplotype for two alleles that are different and are not considered missing by this
+     * HaplotypeConverter.
+     *
+     * Behaviour may be unexpected for alleles which do not pass the {@link
+     * #isConsideredMissing(Genotype)} filter in implementations.
+     *
+     * @param allele1 the first allele
+     * @param allele2 the second allele
+     *
+     * @return the allele that configures the haplotype
+     */
+    abstract Allele getHaplotypeAllele(final Allele allele1, final Allele allele2);
+
+    /**
+     * Convert the genotypes in the variant model into haplotypes using {@link
+     * #getHaplotype(Genotype)} for each genotype
+     */
+    @Override
+    public final VariantContext apply(final VariantContext variant) {
+        Utils.nonNull(variant);
+        VariantContextBuilder builder = new VariantContextBuilder(variant);
+        final List<Genotype> genotypes = variant.getGenotypes().stream()
+                .map(this::getHaplotype).collect(Collectors.toList());
+        builder.genotypes(genotypes);
+        return builder.make();
+    }
+
+    /**
+     * Get the haplotype for a single genotype. If the genotype is already an haploid,
+     * return it as it is. If not, first check if it is considered missing by the model ({@link
+     * #isConsideredMissing(Genotype)}) and if it is not, try to get the alleles using {@link
+     * #getHaplotypeAllele(List)}.
+     */
+    @VisibleForTesting
+    final Genotype getHaplotype(final Genotype genotype) {
```

Comment:
If this is visible for testing it is because it should be tested instead of `apply`, because it makes more sense.